packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -115,7 +128,8 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
 
         // Verify the base fee per gas is correct
         uint256 basefee;
-        (basefee, gasExcess) = _calc1559BaseFee(config, l1Height, parentGasUsed);
+        (basefee, config.gasExcess) =
```

Comment:
this won't write to storage, you have to do`baseFeeConfig.gasExcess`

Reply:
I think you can get rid of  `Config memory config = getConfig();`

Reply:
I can get rid of that struct, sure.